tests/e2e/gpu-feature-discovery_test.go

```diff
@@ -43,22 +43,23 @@ var _ = NVDescribe("GPU Feature Discovery", func() {
 	f := framework.NewFramework("gpu-feature-discovery")
 
 	expectedLabelPatterns := k8sLabels{
-		"nvidia.com/gfd.timestamp":       "[0-9]{10}",
-		"nvidia.com/cuda.driver.major":   "[0-9]+",
-		"nvidia.com/cuda.driver.minor":   "[0-9]+",
-		"nvidia.com/cuda.driver.rev":     "[0-9]*",
-		"nvidia.com/cuda.runtime.major":  "[0-9]+",
-		"nvidia.com/cuda.runtime.minor":  "[0-9]+",
-		"nvidia.com/gpu.machine":         ".*",
-		"nvidia.com/gpu.count":           "[0-9]+",
-		"nvidia.com/gpu.replicas":        "[0-9]+",
-		"nvidia.com/gpu.product":         "[A-Za-z_-]+",
-		"nvidia.com/gpu.memory":          "[0-9]+",
-		"nvidia.com/gpu.family":          "[a-z]+",
-		"nvidia.com/mig.capable":         "[true|false]",
-		"nvidia.com/gpu.compute.major":   "[0-9]+",
-		"nvidia.com/gpu.compute.minor":   "[0-9]+",
-		"nvidia.com/sharing.mps.enabled": "[true|false]",
+		"nvidia.com/gfd.timestamp":        "[0-9]{10}",
+		"nvidia.com/cuda.driver.major":    "[0-9]+",
+		"nvidia.com/cuda.driver.minor":    "[0-9]+",
+		"nvidia.com/cuda.driver.rev":      "[0-9]*",
+		"nvidia.com/cuda.runtime.major":   "[0-9]+",
+		"nvidia.com/cuda.runtime.minor":   "[0-9]+",
+		"nvidia.com/gpu.machine":          ".*",
+		"nvidia.com/gpu.count":            "[0-9]+",
+		"nvidia.com/gpu.replicas":         "[0-9]+",
+		"nvidia.com/gpu.sharing-strategy": "[none|mps|time-slicing]",
+		"nvidia.com/gpu.product":          "[A-Za-z_-]+",
+		"nvidia.com/gpu.memory":           "[0-9]+",
+		"nvidia.com/gpu.family":           "[a-z]+",
+		"nvidia.com/mig.capable":          "[true|false]",
+		"nvidia.com/gpu.compute.major":    "[0-9]+",
+		"nvidia.com/gpu.compute.minor":    "[0-9]+",
+		"nvidia.com/sharing.mps.enabled":  "[true|false]",
```

Comment:
Question -- what component is adding this label to the node?

Reply:
Okay I see this was added to GFD in a previous commit: https://github.com/NVIDIA/k8s-device-plugin/blob/main/internal/lm/nvml.go#L149

Question -- do we still need `nvidia.com/sharing.mps.enabled`? AFAIK we use this as the nodeSelector for the MPS control daemon. Is it sufficient to just use `nvidia.com/gpu.sharing-strategy=mps`?

Reply:
The issue is that the `sharing-strategy` label is per-resource. This means that if we enable resource renaming or start allowing MPS with mixed mig mode, the label to trigger off is not known.

I would say that `nvidia.com/sharing.mps.enabled` is akn to `nvidia.com/mig.capable` in this respect.

@klueska I know that you expressed some concerns over the exact label. Do you have any other suggestions for the label? Should we just make it `nvidia.com/mps.capable` to mirror mig and to indicate that the final decision depends on whether a number of replicas greater than 1 is actually selected?

Reply:
`mig.capable` is different than `nvidia.com/sharing.mps.enabled` in that it says that the GPUs on a node are _capable_ of being split into MIG devices -- it doesn't imply that they are in MIG mode or that there are any MIG devices currently configured.

However it is similar in that it is primarily used to signal that the MIG manager should be deployed on the node (whereas we want something to signal that an MPS daemon should be started for this resource type).

I think dropping `nvidia.com/sharing.mps.enabled` and adding a `nvidia.com/mps.capable` seems reasonable.

Just to make sure we are on the same page -- this would get set if there is a `sharing.mps` section in the config, and that would trigger the MPS daemonset to be deployed (similar to MIG manager) and then start one MPS control daemon per resources type that has replicas > 1. Is that correct?

Reply:
To start: It could also be `nvidia.com/mps.enabled` instead of `nvidia.com/mps.capable`.

We currently trigger the `nvidia.com/sharing.mps.enabled` label if the `sharing.mps` section is present **and** at lease one resource pattern has a replica > 1. This controls the deployment of the MPS daemonset.

In this daemonset the pattern is evaluated against actual resources and if any of these have replicas > 1 then a `nvidia-cuda-mps-control` daemon is started for this resource. A further note is that we currently only support a single resource name and as such only start a single daemon. The plan is to extend this once we have more user feedback / experience.

Reply:
I think we should either make it `nvidia.com/mps.capable` to mimic what we have for MIG, or just stick with `nvidia.com/sharing.mps.enabled`. I think that having  `nvidia.com/mig.capable`, but `nvidia.com/mps.enabled` would be confusing, especially since MPS can be layered on top of MIG.

Speaking of, are there any plans for an equivalent label for `timeSlicing`?  I know there is no daemon that needs to be started in this respect, but does it make sense to add for symmetry?

Reply:
Fair point on using `nvidia.com/mps.capable`. 

There are no current plans for `timeSlicing`, but we can add it. There are, for example, some devices that don't support time slicing and adding this label may make sense there. (see https://github.com/NVIDIA/k8s-dra-driver/pull/58).

I can add an issue to create this label as a follow-up.